pkg/mod/module_test.go

```diff
@@ -0,0 +1,59 @@
+package mod
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAdd(t *testing.T) {
+	t.Parallel()
+
+	var testMods Modules
+	testMods.Add(&Module{Name: "modulepath"})
+	assert.Equal(t, 1, len(testMods))
+	assert.Equal(t, &Module{Name: "modulepath"}, testMods[0])
+}
+
+func TestGetByFilename(t *testing.T) {
+	t.Parallel()
+
+	var testMods Modules
+	testMods.Add(&Module{Name: "modulepath"})
+	assert.Equal(t, &Module{Name: "modulepath"}, testMods.GetByFilename("modulepath/filename"))
+	assert.Equal(t, &Module{Name: "modulepath"}, testMods.GetByFilename(".//modulepath/filename"))
+}
+
+func TestGetByFilepathWithoutValidMod(t *testing.T) {
+	t.Parallel()
+
+	var testMods Modules
+	testMods.Add(&Module{Name: "modulepath"})
+	assert.Nil(t, testMods.GetByFilename("another_modulepath/filename"))
+}
+
+func TestGetByFilepathWithNilMods(t *testing.T) {
+	t.Parallel()
+
+	var testMods Modules
+	assert.Nil(t, testMods.GetByFilename("modulepath/filename"))
+}
+
+func TestFind(t *testing.T) {
+	t.Parallel()
+
+	filename := "github.com/anz-bank/sysl/demo/examples/Modules/deps.sysl"
+	mod, err := Find(filename)
+	assert.Nil(t, err)
+	assert.Equal(t, "github.com/anz-bank/sysl", mod.Name)
```

Comment:
what happened to the rest of the filename?

Reply:
`Find` finds go module of the given file. `mod.Name` is the name of related go module. The downloaded go module doesn't know the accurate file path.